tests/components/konnected/test_panel.py

```diff
@@ -373,3 +373,163 @@ def mock_constructor(host, port, websession):
+                        "5": "Digital Sensor",
+                        "6": "Switchable Output",
+                        "out": "Switchable Output",
+                    },
+                    "binary_sensors": [
+                        {"zone": "1", "type": "door"},
+                        {
+                            "zone": "2",
+                            "type": "window",
+                            "name": "winder",
+                            "inverse": True,
+                        },
+                        {"zone": "3", "type": "door"},
+                    ],
+                    "sensors": [
+                        {"zone": "4", "type": "dht"},
+                        {"zone": "5", "type": "ds18b20", "name": "temper"},
+                    ],
+                    "switches": [
+                        {
+                            "zone": "out",
+                            "name": "switcher",
+                            "activation": "low",
+                            "momentary": 50,
+                            "pause": 100,
+                            "repeat": 4,
+                        },
+                        {"zone": "6"},
+                    ],
+                }
+            ),
+        }
+    )
+
+    entry = MockConfigEntry(
+        domain="konnected",
+        title="Konnected Alarm Panel",
+        data=device_config,
+        options={},
+    )
+    entry.add_to_hass(hass)
+    hass.data[panel.DOMAIN] = {
+        panel.CONF_API_HOST: "192.168.1.1",
+    }
+
+    # override get_status to reflect non-pro board
+    mock_panel.get_status.return_value = {
+        "hwVersion": "2.3.0",
+        "swVersion": "2.3.1",
+        "heap": 10000,
+        "uptime": 12222,
+        "ip": "192.168.1.90",
+        "port": 9123,
+        "sensors": [],
+        "actuators": [],
+        "dht_sensors": [],
+        "ds18b20_sensors": [],
+        "mac": "11:22:33:44:55:66",
+        "settings": {},
+    }
+    device = panel.AlarmPanel(hass, entry)
+    await device.async_save_data()
+    await device.async_connect()
+    await device.update_switch("1", 0)
+
+    # confirm the correct api is used
+    # pylint: disable=no-member
+    assert device.client.put_device.call_count == 1
+    assert device.client.put_zone.call_count == 0
+
+    # confirm the settings are sent to the panel
+    # pylint: disable=no-member
+    assert device.client.put_settings.call_args_list[0][1] == {
+        "sensors": [{"pin": "1"}, {"pin": "2"}, {"pin": "5"}],
+        "actuators": [{"trigger": 0, "pin": "8"}, {"trigger": 1, "pin": "9"}],
+        "dht_sensors": [{"poll_interval": 3, "pin": "6"}],
+        "ds18b20_sensors": [{"pin": "7"}],
+        "auth_token": "11223344556677889900",
+        "blink": True,
+        "discovery": True,
+        "endpoint": "192.168.1.1/api/konnected",
+    }
+
+    # confirm the device settings are saved in hass.data
+    assert hass.data[panel.DOMAIN][panel.CONF_DEVICES] == {
```

Comment:
Saving the data like this is just a detail which could change in the future. This isn't something we should assert.

Reply:
Are you referring to asserting that konnected integration data resides in `hass.data`?

Reply:
Yes. :+1:

Reply:
Got it - updating

Reply:
Updated in latest commits.  There is still one reference to hass.data in these tests to get the instance of the AlarmPanel created when the integration is setup.  This is essential to confirm that the data store is formatted correctly - a check I want to keep to ensure that no future code changes in this module break that data structure since it is utilized when setting up the platforms.